builtin/aws/lambda/releaser.go

```diff
@@ -0,0 +1,323 @@
+				}
+			default:
+				step.Update("Error creating function url config: %q, %q", aerr.Code(), aerr.Message())
+				return err
+			}
+		} else {
+			return err
+		}
+	} else {
+		// update state
+		state.Url = *cfo.FunctionUrl
+		step.Update("Created Lambda URL: %q", state.Url)
+	}
+
+	step.Done()
+
+	return nil
+}
+
+func (r *Releaser) Release(
+	ctx context.Context,
+	log hclog.Logger,
+	src *component.Source,
+	ui terminal.UI,
+	dep *Deployment,
+) (*Release, error) {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	// Create our resource manager and create
+	rm := r.resourceManager(log)
+	if err := rm.CreateAll(
+		ctx, log, sg, ui, src,
+		dep,
+	); err != nil {
+		log.Info("Error creating resources", "error", err)
+		return nil, err
+	}
+
+	// Get our function url state to verify
+	fnUrlState := rm.Resource("function_url").State().(*Resource_FunctionUrl)
+	log.Info("Function URL state", "url", fnUrlState.Url)
+	if fnUrlState == nil {
+		return nil, status.Errorf(codes.Internal, "function url state is nil, this should never happen")
+	}
+
+	return &Release{
+		Url:           fnUrlState.Url,
+		FuncArn:       dep.FuncArn,
+		VerArn:        dep.VerArn,
+		ResourceState: rm.State(),
+	}, nil
+}
+
+func (r *Releaser) Destroy(
+	ctx context.Context,
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Logger: log,
+	})
+	if err != nil {
+		return err
+	}
+
+	rm := r.resourceManager(log)
+
+	// Destroy All
+	return rm.DestroyAll(ctx, log, sg, ui, sess)
+}
+
+// ReleaserConfig is the configuration structure for the Releaser.
+type ReleaserConfig struct {
+	// "AWS_IAM" or "NONE"
+	AuthType string `hcl:"auth_type,optional"`
+	// todo(kevinwang): CORS
+}
+
+func (r *Releaser) Status(
+	ctx context.Context,
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) (*sdk.StatusReport, error) {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Gathering health report for lambda url deployment: %q", release.Url)
+	defer s.Done()
+
+	report := sdk.StatusReport{
+		External: true,
+	}
+
+	// check the function url status
+	log.Info("Checking function url status...", "url", release.Url)
+	if resp, err := http.Get(release.Url); err != nil {
+		log.Error("Failed to get a response from the Lambda URL: %s", err)
+		report.Health = sdk.StatusReport_UNKNOWN
+		report.HealthMessage = "Failed to get a response from the Lambda URL"
+	} else {
+		switch resp.StatusCode {
+		case http.StatusOK:
+			log.Info("Lambda URL returned a 200")
+			report.Health = sdk.StatusReport_READY
+			report.HealthMessage = "Lambda URL appears to be healthy"
+			break
+		default:
+			log.Error("Lambda URL returned a non-200 response: %d", resp.StatusCode)
+			report.Health = sdk.StatusReport_DOWN
+			report.HealthMessage = fmt.Sprintf("Lambda URL returned a non-200 response: %d", resp.StatusCode)
+			break
```

Comment:
I don't thinkkkkk `break` is required in switch statements here 🤔

Reply:
Ah, TIL: `switch` `break` semantics differ slightly between [Go](https://go.dev/play/p/OkbAtjsgGWw) and [TypeScript](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAWwIYzACgJSIN4BQixiECAzlIpalCOYgLyIBMADG0SeQO4xQQAFokw065XIRLTSqcgFNWHAFxcZMsmHJwANvIB0OuAHNMAIgASMM9jUyAJvOCoQOqKvXrN2vYZPmAITg4GzsAXwIIgjQMHCA)